src/code/reducers/helpers/challenge-progress.js

```diff
@@ -36,4 +36,23 @@ export function getScoreValue(score){
     default:
       return score >= 2 ? scoreValues.BRONZE : scoreValues.NONE;
   }
-}
\ No newline at end of file
+}
+
+export function getChallengeScores(level, mission, challengeCount, progress) {
```

Comment:
This logic came from `challenge-award.js`, which was using it to generate the old end-mission dialog. It's used in two different components now, so I figured it would be better off in a utility class here.